### PR TITLE
Prevent unselection of actors who cannot be removed

### PR DESCRIPTION
**Pull request by Lainow**
It was possible to delete users from a ticket even when this was supposed to be blocked by the plugin's configuration. 
now the modification is logged in the backend as well as the front end

**Before**
Screencast-from-17-05-2024-11_17_57

**After**
Screencast-from-17-05-2024-11_15_08

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,12 +5,11 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](http://keepachangelog.com/)
 and this project adheres to [Semantic Versioning](http://semver.org/).
 
-## [2.9.5] - 2024-05-06
+## [2.9.5] - 2024-05-17
```
Comment (Rom1-B) [inline CHANGELOG.md:8]: Please do not modify previously released versions.
To create a new block just write:

```suggestion
## [unreleased]
```

We'll adapt the number and date when we deploy the release.

**Comment by stonebuzz** [comment]
Can you rebase from ```main``` ?

File: inc/ticket.class.php [hunk 2]
```diff
@@ -43,11 +43,7 @@ public static function pre_item_update(CommonDBTM $item)
             !empty(array_filter(
                 $item->input['_actors']['assign'] ?? [],
                 fn ($actor) => $actor['itemtype'] == 'Group'
-            ))
-            && (
-                isset($item->input['_from_assignment'])
-                && $item->input['_from_assignment']
-            )
```
Comment (stonebuzz) [inline inc/ticket.class.php:50]: Should be kept to prevent warning about ```undefined index``` (see https://github.com/pluginsGLPI/escalade/issues/182)

**Review by stonebuzz — CHANGES_REQUESTED** [review]

File: CHANGELOG.md [hunk 3]
```diff
@@ -5,6 +5,12 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](http://keepachangelog.com/)
 and this project adheres to [Semantic Versioning](http://semver.org/).
 
+## [unreleased]
+
+### Fixed
+
+- Prevent unselection of actors who cannot be removed
```
Comment (stonebuzz) [inline CHANGELOG.md:12]:
```suggestion
- Fix ```Display delete button``` option
```

**Review by stonebuzz — CHANGES_REQUESTED** [review]

File: js/remove_btn.js.php [hunk 4]
```diff
@@ -129,12 +183,28 @@
...
+   }
+
    $(document).ready(function() {
       // only in ticket form
       if (location.pathname.indexOf('ticket.form.php') > 0
          || location.pathname.indexOf('problem.form.php') > 0
          || location.pathname.indexOf('change.form.php') > 0) {
          $(document).on('glpi.tab.loaded', function() {
+            // requester_form.on('select2:select', handleEvent);
+            // requester_form.on('select2:unselect', handleEvent);
```
Comment (RomainLvr) [inline js/remove_btn.js.php:207]: Remove commented code
```suggestion
```

**Review by AdrienClairembault — APPROVED** [review]

**Review by Rom1-B — APPROVED** [review]

**Review by stonebuzz — APPROVED** [review]

File: CHANGELOG.md (hunk 3 above)
Comment (stonebuzz) [reply CHANGELOG.md]: Fix(Core): Fix behavior of 'Display delete button' option

**Pull request merged**